examples/nodejs/rate-limiter/README.md

```diff
@@ -0,0 +1,109 @@
+
+To get started with the rate-limiter:
+- You will need a Momento API key. You can obtain one from the [Momento Console](https://console.gomomento.com).
+- You will need to create a cache called `rate-limiter` from the console as well!
+
+Once you have the key and the cache created, you can begin integration! Remember to store your API key in an environment variable named `MOMENTO_API_KEY`.
+
+```typescript
+const momento = await CacheClient.create({
+  configuration: Configurations.Laptop.v1(),
+  credentialProvider: CredentialProvider.fromEnvironmentVariable({
+    environmentVariableName: "MOMENTO_API_KEY",
+  }),
+  defaultTtlSeconds: 6000,
+});
+
+const tpmLimit = 10;
+const rateLimiter = new MomentoRateLimiter(momento, tpmLimit);
+
+// test rate limiter
+const limitExceeded : boolean = await rateLimiter.isLimitExceeded(`id`);
+```
+
+## Key Mechanism
+
+At the heart of our rate-limiter is a key mechanism that allows us to perform rate limiting based on user-per-minute granularity. The key is constructed using a combination of the user ID and the current minute. This key plays a pivotal role in tracking and limiting the number of transactions a user can make in a given minute.
+
+```typescript
+generateMinuteKey(baseKey: string): string {
+    const currentDate = new Date();
+    const currentMinute = currentDate.getMinutes();
+    return `${baseKey}_${currentMinute}`;
+  }
+```
+
+## Running the example
+
+To run the example, you'll need:
+
+- Node version 14 or higher
+- A Momento API key, which you can obtain from the [Momento Console](https://console.gomomento.com).
+
+```bash
+npm install
+MOMENTO_API_KEY="yourApiKey" npm run rate-limiter
+```
+
+You'll see a sample output that contains statistics about the rate-limiter.
+
+```bash
+
+Simulating 1000 requests for each rate limiter with a random delay between requests upto a max of 60000 milliseconds.
+
+Momento Rate Limiter - Successes: 1000
+Momento Rate Limiter - Throttles: 0
+Momento Rate Limiter - Errors: 0
+Momento Rate Limiter - Average Latency: 2.427
+Momento Rate Limiter - p50 Latency: 2
+Momento Rate Limiter - p90 Latency: 3
+Momento Rate Limiter - p99 Latency: 5
+Momento Rate Limiter - p99.9 Latency: 34
+
+All tasks complete!
+
+```
+
+There are three additional arguments that you can provide to the rate-limiter if you want to experiment with different configurations:
+
+- totalRequests: The total number of requests that the example will simulate, defaulted to 1000.
+- randomDelayUpperBound: The simulation adds a random delay between 0 and the randomDelayUpperBound, defaulted to 60 seconds.
+- tpmLimit: The rate per minute at which a user or entity will be allowed by the rate-limiter, defaulted to 500.
+
+To override totalRequests to 10, randomDelayUpperBound to 60, and tpmLimit to 1, the command will look like:
+
+```bash
+MOMENTO_API_KEY="yourApiKey" npm run rate-limiter 10 60 1
```

Comment:
maybe add one more sentence after this that says "if you want to see some throttles, try these settings". And optionally paste another example of your output where throttle count is not zero.